src/components/App.tsx

```diff
@@ -29,29 +36,54 @@ type GetArticlesParams = {
 
 // A types object for thr Article that can be easily passed and used where needed
 export interface Article {
-  source: Source;
-  author: null;
+  author: string;
   title: string;
   description: string;
   url: string;
   urlToImage: string;
   publishedAt: string;
-  content: string;
+  _id: string;
 }
 
-export interface Source {
-  id: string;
+type loginProps = {
+  email: string;
+  password: string;
+};
+
+type signupProps = {
+  name: string;
+  avatar: string;
+  email: string;
+  password: string;
+};
+
+export interface currentUser {
   name: string;
+  avatar: string;
+  email: string;
+  password: string;
 }
 
+type updateUserProps = {
+  name: string;
+  avatar: string;
+};
+
 function App() {
   const [activeModal, setActiveModal] = useState("");
-  const [isLoggedIn] = useState(true);
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [currentUser, setCurrentUser] = useState(() => {
+    const storedUser = localStorage.getItem("currentUser");
+    return storedUser ? JSON.parse(storedUser) : null;
+  });
   const [cardsData, setCardsData] = useState<Article[]>([]);
   const [searchedArticles, setSearchedArticles] = useState(false);
   const [isLoading, setIsLoading] = useState(false);
   const [searchResults, setSearchResults] = useState(false);
   const [ArticlesError, setArticlesError] = useState("");
```

Comment:
`ArticlesError`

Every variable in the project should be in `camelCase` (in **lower case** if it's not a class or component)